packages/ckeditor5-engine/src/model/differ.ts

```diff
@@ -1231,7 +1264,7 @@ export default class Differ {
 
 		for ( const item of range.getItems( { shallow: true } ) ) {
 			if ( item.is( 'element' ) ) {
-				this._elementSnapshots.delete( item );
+				this._elementChildrenSnapshots.delete( item );
```

Comment:
It cannot be done because `rename` consist `remove` and `insert` operations [1]. `remove` operation calls `_removeAllNestedChanges` method, which removes a snapshot of the element used in `before` attribute. After removing such snapshot `before` in `rename` operation is undefined which misses the whole point of this ticket. 

[1] 
```ts
			case 'rename': {
				if ( this._isInInsertedElement( operation.position.parent ) ) {
					return;
				}

				this._markRemove( operation.position.parent, operation.position.offset, 1, 'rename' );
				this._markInsert( operation.position.parent, operation.position.offset, 1, 'rename' );

				const range = Range._createFromPositionAndShift( operation.position, 1 );

				for ( const marker of this._markerCollection.getMarkersIntersectingRange( range ) ) {
					const markerData = marker.getData();

					this.bufferMarkerChange( marker.name, markerData, markerData );
				}

				break;
			}
```

Reply:
Ah, okay, thanks, I see what is going on here. :+1: